common/txmgr/inmemory_store.go

```diff
@@ -315,6 +317,55 @@ func (ms *inMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) SaveC
 	return nil
 }
 func (ms *inMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) SaveInProgressAttempt(ctx context.Context, attempt *txmgrtypes.TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) error {
+	ms.addressStatesLock.RLock()
+	defer ms.addressStatesLock.RUnlock()
+
+	var tx *txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]
+	for _, as := range ms.addressStates {
+		fn := func(tx *txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) bool {
+			return true
+		}
+		txs := as.findTxs(nil, fn, attempt.TxID)
+		if len(txs) != 0 {
+			tx = &txs[0]
+			break
+		}
+	}
```

Comment:
Similar to the comment in other PRs, couldn't we use the tx in the attempt to get the address state for the right `FromAddress` without having to loop over them? `attempt.Tx.FromAddress`